### PR TITLE
Fixed expanding ONNX function

### PR DESCRIPTION
**Pull request by mbencer**
Ticket: 62895, 65805, 38726
1. Fixed expanding ONNX function:
image

Note that an ONNX function can be inside another ONXN Function. 

2. Default `PartialShape` ctor which produces a scalar. If an input shape is not provided `PartialShape::dynamic()` should be set.

### CODE REVIEW

File: ngraph/frontend/onnx/frontend/src/core/transform.cpp [hunk 1]
```diff
@@ -38,19 +50,29 @@ ONNX_NAMESPACE::TypeProto get_input_type(std::string const& name, ONNX_NAMESPACE
...
 
-    for (int i = 0; i < graph_proto->node().size(); ++i) {
-        ONNX_NAMESPACE::NodeProto node = graph_proto->node().Get(i);
+    ONNX_NAMESPACE::GraphProto new_graph;
+    int origin_graph_index = 0;
+    int new_graph_index = 0;
+    while (origin_graph_index < graph_proto->node().size() || new_graph_index < new_graph.node().size()) {
+        ONNX_NAMESPACE::NodeProto* node = nullptr;
+        if (new_graph_index < new_graph.node().size()) {
+            // nodes expenaded from function can have another functions
```
Comment (t-jankowski) [inline ngraph/frontend/onnx/frontend/src/core/transform.cpp:59]: Like spelling mistake  it looks.

File: ngraph/frontend/onnx/frontend/src/core/transform.cpp [hunk 2]
```diff
@@ -38,19 +50,29 @@ ONNX_NAMESPACE::TypeProto get_input_type(std::string const& name, ONNX_NAMESPACE
 void ngraph::onnx_import::transform::expand_onnx_functions(ONNX_NAMESPACE::ModelProto& model_proto) {
     auto graph_proto = model_proto.mutable_graph();
 
-    for (int i = 0; i < graph_proto->node().size(); ++i) {
-        ONNX_NAMESPACE::NodeProto node = graph_proto->node().Get(i);
+    ONNX_NAMESPACE::GraphProto new_graph;
+    int origin_graph_index = 0;
+    int new_graph_index = 0;
+    while (origin_graph_index < graph_proto->node().size() || new_graph_index < new_graph.node().size()) {
```
Comment (tomdol) [inline ngraph/frontend/onnx/frontend/src/core/transform.cpp:56]: You can consider using https://en.cppreference.com/w/cpp/algorithm/rotate to move all new added nodes to the correct position.

**Review by tomdol — APPROVED** [review]
Just wondering - perhaps one more unit test proving the correct ordering of nodes would be useful here?

File: ngraph/frontend/onnx/frontend/src/core/transform.cpp [hunk 3]
```diff
@@ -38,19 +50,29 @@ ONNX_NAMESPACE::TypeProto get_input_type(std::string const& name, ONNX_NAMESPACE
...
+        ONNX_NAMESPACE::NodeProto* node = nullptr;
+        if (new_graph_index < new_graph.node().size()) {
+            // nodes expenaded from function can have another functions
+            node = new_graph.mutable_node(new_graph_index++);
+        } else {
+            node = new_graph.add_node();
+            node->Swap(graph_proto->mutable_node(origin_graph_index++));
+        }
```
Comment (t-jankowski) [inline ngraph/frontend/onnx/frontend/src/core/transform.cpp:64]: Does it iterate twice over the graph without function ops?

**Review by t-jankowski — CHANGES_REQUESTED** [review]

**Comment by mbencer** [comment]
> Just wondering - perhaps one more unit test proving the correct ordering of nodes would be useful here?

Added new tests to `openvino/ngraph/test/onnx/onnx_transformations.cpp`

**Pull request merged**